src/software/ai/navigator/path_planner/path_planner_performance_test.cpp

```diff
@@ -0,0 +1,224 @@
+     .obstacles      = {},
+     .num_iterations = 10},
+
+    {.name           = "Empty divB area long path",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = {},
+     .num_iterations = 10},
+
+    {.name           = "10 obstacles on full field short path",
+     .start          = Point(-2, 0),
+     .end            = Point(2, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_10,
+     .num_iterations = 10},
+
+    {.name           = "20 obstacles on full field short path",
+     .start          = Point(-2, 0),
+     .end            = Point(2, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_20,
+     .num_iterations = 10},
+
+    {.name           = "30 obstacles on full field short path",
+     .start          = Point(-2, 0),
+     .end            = Point(2, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_30,
+     .num_iterations = 10},
+
+    {.name           = "10 obstacles on full field",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_10,
+     .num_iterations = 10},
+
+    {.name           = "20 obstacles on full field",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_20,
+     .num_iterations = 10},
+
+    {.name           = "30 obstacles on full field",
+     .start          = Point(-4.5, 0),
+     .end            = Point(4.5, 0),
+     .navigable_area = ::Test::TestUtil::createSSLDivBField().fieldBoundary(),
+     .obstacles      = obstacles_30,
+     .num_iterations = 10},
+};
+
+template <typename PlannerT>
+std::pair<std::string, PathPlannerConstructor> name_and_constructor()
+{
+    return std::pair<std::string, PathPlannerConstructor>(
+        typeid(PlannerT).name(), []() { return std::make_unique<PlannerT>(); });
+}
+
+std::vector<std::pair<std::string, PathPlannerConstructor>>
+    path_planner_names_and_constructors = {
+        // add path planner constructors here
+        name_and_constructor<ThetaStarPathPlanner>(),
+        name_and_constructor<StraightLinePathPlanner>()};
+
+class PlannerPerformanceTest
+    : public testing::TestWithParam<
+          std::tuple<std::pair<std::string, PathPlannerConstructor>, PlannerTestCase>>
+{
+};
+
+// This test is disabled, it can be enabled by removing "DISABLED_" from the test name
+TEST_P(PlannerPerformanceTest, DISABLED_path_planner_performance)
+{
+    std::string name                     = std::get<0>(GetParam()).first.substr(2);
```

Comment:
What does the substr cut out?

Reply:
it strips out a couple digits from the beginning